### DISCUSSION TITLE
Hide Attach button depending on the relationship

### DISCUSSION THREAD

**Discussion by DarkCoderJ** [filamentphp/filament#5217] [Help] [answered]
I have this two relationships

<img width="944" alt="Liga" src="https://user-images.githubusercontent.com/113643262/206771747-4b9eb007-6de1-4aa4-a6a4-a8dffc3e536a.png">

<img width="952" alt="Player" src="https://user-images.githubusercontent.com/113643262/206771823-e3f8d394-64ec-442d-b2d7-29335cf2d2f3.png">

Is it possible to hide the attach button only in the second case?

Thank you.

**Comment by Morry981** [comment, marked as answer]
Hi, hope this help.

You can make it visible if ownerRecord pass your condition
https://filamentphp.com/docs/2.x/tables/actions#authorization
https://filamentphp.com/docs/2.x/admin/resources/relation-managers#accessing-the-owner-record
```
->headerActions([
    Tables\Actions\AttachAction::make()->visible(
        fn(RelationManager $livewire): bool => $livewire->ownerRecord->name === "Lapidarios"
    ),
 ]);
```
Same for actions() and bulkActions() items i think

EDIT:
or probably better, override method
https://github.com/filamentphp/filament/discussions/82#discussioncomment-454866
```
protected function canAttach(): bool
{
    return $this->ownerRecord->name === "Lapidarios";
}
```


I tried with policies as said in https://filamentphp.com/docs/2.x/admin/resources/creating-records#authorization
something like Gate::define("attach", fn(User .., Model ..)...);
but i do not find out why it not pass as second argument the model (maybe i open another discussion)

Let me know

**Reply by DarkCoderJ** [reply]
This worked:
`->headerActions([
    Tables\Actions\AttachAction::make()->visible(
        fn(RelationManager $livewire): bool => $livewire->ownerRecord->name === "Lapidarios"
    ),
 ]);
`

I have made a change to make it work for all objects of the class
`->hidden( fn(RelationManager $livewire)=> get_class($livewire->ownerRecord) === Jugador::class)`

Thank you!!!
